includes/malloc.h

```diff
@@ -0,0 +1,31 @@
+#ifndef MALLOC_H
+#define MALLOC_H
+
+#include <stdint.h>
+#include <unistd.h>
+
+extern uint32_t _ebss;
+
+#define HEAP_START ((void*)&_ebss);
+
+#define ALIGN(x) (((((x) - 1) >> 2) << 2) + 4)
+
+#define HEADER_SIZE (sizeof(size_t) * 2 + sizeof(int) + sizeof(void*) * 2)
+
+#define RAM_SIZE 32000
```

Comment:
Is it RAM_SIZE or HEAP_SIZE? HEAP_SIZE should be smaller, and you could use some linker script magic to calculate the heap size. (Or just take end of ram address and subtract heap_start from it.)